FX2024 Domain & Capabilities update /Quantify Business Value / Benchmarking (#646)
microsoft/finops-toolkit: docs/_docs/framework/capabilities/quantify/benchmarking.md

grand_parent: FinOps Framework
parent: Quantify
title: Benchmarking
permalink: /framework/capabilities/quantify/benchmarking
nav_order: 3
description: This article helps you understand the benchmarking capability within the FinOps Framework and how to implement that in the Microsoft Cloud.
author: ripadrao
ms.author: banders
ms.date: 03/25/2024
ms.topic: conceptual
ms.service: finops
ms.reviewer: micflan
---

<span class="fs-9 d-block mb-4">Benchmarking</span>
This article helps you understand the benchmarking capability within the FinOps Framework and how to implement that in the Microsoft Cloud.
{: .fs-6 .fw-300 }

<details open markdown="1">
  <summary class="fs-2 text-uppercase">On this page</summary>

- [▶️ Getting started](#️-getting-started)
- [🏗️ Building on the basics](#️-building-on-the-basics)
- [🍎 Learn more at the FinOps Foundation](#-learn-more-at-the-finops-foundation)
- [⏩ Next steps](#-next-steps)
- [🧰 Related tools](#-related-tools)

</details>

---

<a name="definition"></a>
**Benchmarking is a systematic process of evaluating the performance and value of cloud services using efficiency metrics, either within an organization or against industry peers.**

> 

Reviewer: ```suggestion
```

{: .fs-6 .fw-300 }

> Benchmarking enables organizations to understand how key performance indicators (KPIs) compare with others and identify areas to improve and possibly learn from. Benchmarking encompasses internal evaluations across various teams within the organization, as well as external comparisons with peer entities utilizing Microsoft Cloud services in similar capacities. It's important to recognize that cloud usage is unique to each organization, and there isn't a single "correct" approach when comparing KPIs across teams and organizations.

Reviewer: ```suggestion
Identify and automate key performance indicators (KPIs) based on organizational priorities. Compare across internal teams and divisions or other companies, when possible, to identify areas to improve and possibly learn from others. Remember that cloud usage is unique to each organization, and there isn't a single "correct" approach when comparing KPIs across teams and organizations.
```

> Benchmarking serves as a tool to gauge performance, encouraging companies to make informed decisions based on current knowledge rather than deferring action in anticipation of better data. Establish well-defined metrics, maintain transparent communication regarding goals and objectives, ensure precise data collection and effective dashboarding, and garner management support to maximize your return on investment from benchmarking efforts.

Reviewer: ```suggestion
Leverage benchmarking as a tool to measure performance and progress against organizational goals. Encourage teams to mature and make informed decisions based on available data rather than deferring in anticipation of "better" data. Establish well-defined metrics, maintain transparent communication regarding goals and objectives, ensure precise data collection and effective dashboarding, and garner management support to maximize your return on investment from benchmarking efforts.
```

<br>

## ▶️ Getting started

> When you first start managing cost in the cloud, leverage the existing guidance and recommendations which are based on benchmarks established across all Microsoft Cloud customers, such as the [Azure Advisor score](https://learn.microsoft.com/azure/advisor/azure-advisor-score) and the [well-architected assessment](https://learn.microsoft.com/assessments/azure-architecture-review/).

Reviewer: ```suggestion
When you first start managing cost in the cloud, leverage the existing guidance and recommendations which are based on benchmarks established across all Microsoft Cloud customers:
```

> - Start by reviewing the [Azure Advisor score](https://learn.microsoft.com/azure/advisor/azure-advisor-score) at the primary scope you manage, whether that's a subscription, resource group, or based on tags.

Reviewer: ```suggestion
- Review the [Azure Advisor score](https://learn.microsoft.com/azure/advisor/azure-advisor-score) at the primary scope you manage, whether that's a subscription, resource group, or based on tags.
```

  - The Advisor score consists of an overall score, which can be further broken down into five category scores. One score for each category of Advisor represents the five pillars of the Well-Architected Framework.

>   - Leverage the [Workload optimization](../optimize/workloads.md] capability to prioritize and implement recommendations with the highest priority.
>   - Leverage the [Rate optimization](../optimize/commitment-discounts.md) capability to maximize savings with commitment discounts, like reservations and savings plans.

Reviewer: ```suggestion
```

- Complete the [Azure Well-Architected Review self-assessment](https://learn.microsoft.com/azure/well-architected/cross-cutting-guides/implementing-recommendations) to identify areas your existing workloads can be improved based on the Azure Well-Architected Framework.

>     - Link your subscription to include Azure Advisor recommendations in the assessment.

Reviewer: ```suggestion
  - Link your subscription to include Azure Advisor recommendations in the assessment.
```

  - Total cost impact of idle resources each month or quarter.
  - Response time to detect and resolve anomalies.
  - Number of false positives and false negatives.
- Build and share reports covering your KPIs to publicize benchmarks within the organization.
- Foster a culture of continuous learning, innovation, and collaboration by celebrating successes and sharing proven practices.
  - Regularly review and refine the benchmarking baseline based on feedback, industry best practices, and emerging technologies.
  - Promote knowledge sharing and cross-functional collaboration to drive continuous improvement in the benchmarking capability.

<br>

## 🍎 Learn more at the FinOps Foundation

This capability is a part of the FinOps Framework by the FinOps Foundation, a non-profit organization dedicated to advancing cloud cost management and optimization. For more information about FinOps, including useful playbooks, training and certification programs, and more, see to the [benchmarking](https://www.finops.org/framework/capabilities/benchmarking) article in the FinOps Framework documentation.

You can also find related videos on the FinOps Foundation YouTube channel:

> <!--[!VIDEO https://www.youtube.com/embed/{id}?list={list}]-->
> {% include video.html title="Budgeting videos" id="5Qe7eRXKMRzRrwBI" list="PLUSCToibAswnjB7fYRA02ePxySkpDex6q" %}
> 

Reviewer: ```suggestion
<!--
[!VIDEO https://www.youtube.com/embed/{id}?list={list}]
{% include video.html title="Budgeting videos" id="5Qe7eRXKMRzRrwBI" list="PLUSCToibAswnjB7fYRA02ePxySkpDex6q" %}
-->

```

<br>

## ⏩ Next steps

Related FinOps capabilities:

- [Forecasting](./forecasting.md)
- [Budgeting](./budgeting.md)

> - [Unit economics](./unit-economics.md)

Reviewer: ```suggestion
- [Unit economics](./unit-economics.md)
- [Workload optimization](../optimize/workloads.md)
- [Rate optimization](../optimize/rate-optimization.md)
```

  - Prioritize the recommendations with high priorities. 
  - If any of the Advisor recommendation related to Reserved Instances or Savings Plan for compute, refer to [commitment-based](docs/_docs/framework/capabilities/optimize/commitment-discounts.md) to learn more and about this capability.
- Continue evaluating the health of your environment by compliting an [Azure Well-Architected Review assessment](https://learn.microsoft.com/en-us/azure/well-architected/cross-cutting-guides/implementing-recommendations) which is a self-assessment that can help a workload team examine a workload from the perspective of the Azure Well-Architected Framework. 
    - Link your Azure Subscription to allow Azure Advisor recommendations to be included in this assessment.

<br>

## 🏗️ Building on the basics

Build up on what you already know about Well-Architecture Framework. Define internally your benchmarking metrics, communicate and manage different benchmarking programs for the different teams. 

> - Establish and automate KPIs, such as:

Reviewer 2: Created this document in the past which can give some ideas on KPI's to track. Should we mention any of these here? 
(example: number of idle resources per workload)

[FinOps Technical Implementation - OKRs.docx](https://github.com/user-attachments/files/15842820/FinOps.Technical.Implementation.-.OKRs.docx)

> - Establish and automate KPIs, such as:

Reviewer: I'd love to add these to the capability docs. These and the ones the Foundation has. Next step beyond that would be to offer scripts, reports, etc. to automate the metrics.

> - Establish and automate KPIs, such as:

Reviewer: Resolving this so we can close this PR. @KevDLR, let's chat about how we can flesh these out after FinOps X.

  - Number of anomalies each month or quarter.
  - Total cost impact of idle resources each month or quarter.
  - Response time to detect and resolve anomalies.
  - Number of false positives and false negatives.
- Expand coverage of anomaly detection and response processes to include all costs.

> - Foster a culture of continuous learning, innovation, and collaboration.

Reviewer: ```suggestion
- Build and share reports covering your KPIs to publicize benchmarks within the organization.
- Foster a culture of continuous learning, innovation, and collaboration by celebrating successes and sharing proven practices.
```

  - Regularly review and refine the benchmarking baseline based on feedback, industry best practices, and emerging technologies.
  - Promote knowledge sharing and cross-functional collaboration to drive continuous improvement in the benchmarking capability.

<br>

## 🍎 Learn more at the FinOps Foundation

> This capability is a part of the FinOps Framework by the FinOps Foundation, a non-profit organization dedicated to advancing cloud cost management and optimization. For more information about FinOps, including useful playbooks, training and certification programs, and more, see to the [benchmarking](https://www.finops.org/framework/capabilities/benchmarking) article in the FinOps Framework documentation.
> 
> 
> 
> You can also find related videos on the FinOps Foundation YouTube channel:

Reviewer: ```suggestion
This capability is a part of the FinOps Framework by the FinOps Foundation, a non-profit organization dedicated to advancing cloud cost management and optimization. For more information about FinOps, including useful playbooks, training and certification programs, and more, see to the [benchmarking](https://www.finops.org/framework/capabilities/benchmarking) article in the FinOps Framework documentation.

You can also find related videos on the FinOps Foundation YouTube channel:
```

<!--[!VIDEO https://www.youtube.com/embed/{id}?list={list}]-->
{% include video.html title="Budgeting videos" id="5Qe7eRXKMRzRrwBI" list="PLUSCToibAswnjB7fYRA02ePxySkpDex6q" %}

<br>

## ⏩ Next steps

Related FinOps capabilities:

- [Forecasting](./forecasting.md)

> - [Onboarding workloads](../manage/onboarding.md)
> - [Chargeback and finance integration](../manage/invoicing-chargeback.md)

Reviewer: ```suggestion
- [Budgeting](./budgeting.md)
- [Unit economics](./unit-economics.md)
```
